### PR TITLE
Feature/add diff

### PR DESCRIPTION
**Pull request by samjacoby**
Failure similar to the show function — but gives a diff between two commits. Added tests, which pass and seem to work fine. 

Plays nicely w/ adventure, which uses it to show diff between head and a specific sha. 

If you get a chance, @albertsun, could you take a look?

### CODE REVIEW

File: lib/capital_git.rb [hunk 1]
```diff
@@ -30,7 +30,8 @@ def self.base_keypath
   # defined in a config file
   def self.load! path_to_config, environment = nil
     environment = self.env_name if environment.nil?
-
+    puts environment
+    puts File.read(path_to_config)
```
Comment (blumysden) [inline lib/capital_git.rb]: We should yank these

**Comment by blumysden** [comment]
Excellent.  Thanks, Sam.

File: lib/capital_git.rb (hunk 1 above)
Comment (samjacoby) [inline lib/capital_git.rb]: Whoops — yes. done.

**Comment by albertsun** [comment]
Yep, will take a look a bit later today

**Comment by samjacoby** [comment]
OK — I am going to write something else to read out a file at a specific
sha, so I'll put that in here also when I finish.

**Comment by albertsun** [comment]
The `read` method currently supports reading a file at a specific branch by passing `options[:branch]` — I think it would make sense to support a similar API for reading at a specific SHA by passing `options[:sha]` or something like that — and then having the format of the `options` object be consistent across methods.

**Comment by samjacoby** [comment]
Yes! That's exactly what I'm implementing.

**Comment by albertsun** [comment]
So looking at the method signature for this and show, thinking we might want to more closely match git's command line arguments by passing commit id's as optional arguments instead of as options.

How about

`diff(sha)` Returns diff between `sha` and HEAD
`diff(sha1, sha2)` Returns diff between `sha1` and `sha2`

Then (even though this PR doesn't change show) let's change show to have this signature

`show()` Returns diff between HEAD and HEAD~1
`show(sha)` Returns diff between `sha` and it's parent.

**Comment by samjacoby** [comment]
OK. That would be my intuition, but I was matching the options pattern.
That's also what rugged expects, more or less — passing around `nil` for
whatever you're not interested in.

**Comment by samjacoby** [comment]
I modified by the read() method as well — there's no real clean analogue
for that in git, I guess, but at the moment, it also takes sha as an
option.

**Comment by albertsun** [comment]
Cool. Looks good!

**Pull request merged**